vm/src/types/instance_definitions/builtins_instance_def.rs

```diff
@@ -192,25 +194,54 @@ impl BuiltinsInstanceDef {
         }
     }
 
-    pub(crate) fn dynamic() -> BuiltinsInstanceDef {
+    pub(crate) fn dynamic(params: CairoLayoutParams) -> BuiltinsInstanceDef {
+        let pedersen = Some(PedersenInstanceDef {
+            ratio: Some(params.pedersen_ratio),
+        });
+        let range_check = Some(RangeCheckInstanceDef {
+            ratio: Some(params.range_check_ratio),
+        });
+        let ecdsa = Some(EcdsaInstanceDef {
+            ratio: Some(params.ecdsa_ratio),
+        });
+        let bitwise = Some(BitwiseInstanceDef {
+            ratio: Some(params.bitwise_ratio),
+        });
+        let ec_op = Some(EcOpInstanceDef {
+            ratio: Some(params.ec_op_ratio),
+        });
+        let keccak = Some(KeccakInstanceDef {
+            ratio: Some(params.keccak_ratio),
+        });
+        let poseidon = Some(PoseidonInstanceDef {
+            ratio: Some(params.poseidon_ratio),
+        });
```

Comment:
Shouldn't we check the ratios?
If the ratio is 0 then it means the builtin shouldn't be included in the layout

Reply:
I tried to copy cairo-lang as best as I could, in that case, cairo lang adds the builtins with ratio 0. https://github.com/starkware-libs/cairo-lang/blob/a86e92bfde9c171c0856d7b46580c66e004922f3/src/starkware/cairo/lang/instances.py#L114-L128. Is that equivalent to not adding it at all?

Reply:
Oh, I was looking at this [method ](https://github.com/starkware-libs/cairo-lang/blob/a86e92bfde9c171c0856d7b46580c66e004922f3/src/starkware/cairo/lang/dynamic_layout_params.py#L71) which has that logic, but it it not the one used by cairo_run to create the dynamic layout, you can ignore this comment!